src/main/java/health/ere/ps/service/cetp/SubscriptionManager.java

```diff
@@ -59,28 +67,171 @@ public class SubscriptionManager {
 
     private volatile boolean configsLoaded = false;
 
+    private ExecutorService threadPool;
+
     void onStart(@Observes StartupEvent ev) {
         loadKonnektorConfigs();
-
+        threadPool = Executors.newFixedThreadPool(hostToKonnektorConfig.size());
         SubscriptionsMXBeanImpl subscriptionsMXBean = new SubscriptionsMXBeanImpl(hostToKonnektorConfig.size());
         PsMXBeanManager.registerMXBean(SubscriptionsMXBean.OBJECT_NAME, subscriptionsMXBean);
+    }
 
-        renewSubscriptions();
+    @Scheduled(
```

Comment:
I don't think it is a good idea to renew the subscription every 3 seconds.

Reply:
Three will be a time window around 50ms - 300ms where no subscription is available and this can be a problem during high load time.
I think we should observe the load and only to a resubscribe if we did not have any messages in the last minute.

Reply:
We only check every 3 seconds, actual renewal will happen if termintation time is close
```
int safePeriod = appConfig.getCetpSubscriptionsRenewalSafePeriodMs();
if (now.getTime() + safePeriod >= expireDate.getTime())
```
Also if no active subscription found on Konnektor side we must do `subscribe`